vif_plug_vhostuser/linux_net.py

```diff
@@ -23,59 +21,31 @@
 LOG = logging.getLogger(__name__)
 
 
-def _ovs_vsctl(args, timeout=timeout):
+def _ovs_vsctl(args, timeout=30):
     full_args = ['ovs-vsctl', '--timeout=%s' % timeout] + args
     try:
         return processutils.execute(*full_args, run_as_root=True)
     except Exception as e:
         LOG.error(_LE("Unable to execute %(cmd)s. Exception: %(exception)s"),
                   {'cmd': full_args, 'exception': e})
-        raise exception.AgentError(method=full_args)
+        raise e.AgentError(method=full_args)
 
 
-def create_ovs_vif_port(bridge, dev, iface_id, mac, instance_id, mtu,
-                        timeout=timeout):
-    _ovs_vsctl(['--', '--if-exists', 'del-port', dev, '--',
-                'add-port', bridge, dev,
-                '--', 'set', 'Interface', dev,
-                'external-ids:iface-id=%s' % iface_id,
-                'external-ids:iface-status=active',
-                'external-ids:attached-mac=%s' % mac,
-                'external-ids:vm-uuid=%s' % instance_id],
-                timeout=timeout)
-    _set_device_mtu(dev, mtu)
+def create_ovs_vif_port(bridge, dev, iface_id, mac, instance_id,
+                        timeout=30, type=None):
+    cmd = ['--', '--if-exists', 'del-port', dev, '--',
+           'add-port', bridge, dev,
+           '--', 'set', 'Interface', dev,
+           'external-ids:iface-id=%s' % iface_id,
+           'external-ids:iface-status=active',
+           'external-ids:attached-mac=%s' % mac,
+           'external-ids:vm-uuid=%s' % instance_id]
+    if type:
+        cmd += ['type=%s' % type]
 
-
-def delete_ovs_vif_port(bridge, dev, timeout=timeout):
-    _ovs_vsctl(['--', '--if-exists', 'del-port', bridge, dev],
-               timeout=timeout)
-    delete_net_dev(dev)
```

Comment:
yes in the case of vhost-user as it is a userspace interface there is no tap device to delete via the ip command. the acll to del-port will remove the vhost-user socket.